backends/nxp/tests/test_quantizer.py

```diff
@@ -0,0 +1,204 @@
+# Copyright 2024 NXP
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+# Tests for NeutronQuantizer.
+
+import torch
+from torch.ao.quantization.quantize_pt2e import prepare_pt2e, convert_pt2e
+from torch.fx import GraphModule
+
+from executorch.backends.nxp.quantizer.neutron_quantizer import NeutronQuantizer
+import executorch.backends.nxp.tests.models as models
+
+
+def _get_target_name(node):
+    return node._pretty_print_target(node.target)
+
+
+def test_quantizer_conv2d():
+    model = models.Conv2dModule()
+    model.eval()
+
+    example_input = (torch.ones(1, 4, 32, 32),)
+    quantizer = NeutronQuantizer()
+    graph_module = torch.export.export_for_training(model, example_input, strict=True).module()
+
+    # noinspection PyTypeChecker
+    m = prepare_pt2e(graph_module, quantizer)
+    m(*example_input)
+    m = convert_pt2e(m)
+    # print(f"Quantized model: {m}")
+    nodes = list(m.graph.nodes)
+
+    assert len(nodes) == 11
+    assert nodes[7].name == "conv2d"
+    # [0]: Input, [1] : weights, [2]: bias
+    assert _get_target_name(nodes[7].args[0]) == 'torch.ops.quantized_decomposed.dequantize_per_tensor.default'
```

Comment:
Other option would be to use `FileCheck` utils, not a strong preference.

Reply:
Thanks for the tip. We plan to utilize `FileCheck` through some `ArmTester`-like interface, but we're not there yet. `FileCheck` in general doesn't provide an API to check order and specific index of an operators, what is at this stage important for us, because we're (not ideally) using models with multiple operators in tests.

Reply:
`ArmTester` derived from `XNNPACKTester`, alas same story as Quantizer. But since you don't have it implemented yet, can you please refactor `XNNPACKTester`, abstract parts of it as `BackendTester` and move of it to `backends/test` dir and make `XNNPACKTester`, `ArmTester` (already inherits from `XNNPACKTester`), and `NeutronTester` (new!) derive from that common tester? 🙏

Reply:
Sounds good to me. Should I open Github issue or we can track this internally?

Reply:
Please do open a Github issue to track

Reply:
https://github.com/pytorch/executorch/issues/10100